x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/stats/SearchContextStats.java

```diff
@@ -292,6 +292,24 @@ private boolean detectSingleValue(IndexReader r, MappedFieldType fieldType, Stri
         return false;
     }
 
+    @Override
+    public boolean canUseEqualityOnSyntheticSourceDelegate(String name, String value) {
+        for (SearchExecutionContext ctx : contexts) {
+            MappedFieldType type = ctx.getFieldType(name);
+            if (type == null) {
+                return false;
+            }
+            if (type instanceof TextFieldMapper.TextFieldType t) {
+                if (t.canUseSyntheticSourceDelegateForQueryingEquality(value) == false) {
+                    return false;
+                }
+            } else {
+                return false;
+            }
+        }
+        return true;
+    }
```

Comment:
Should this be part of `FieldConfig`?

Reply:
I don't *think* so. Field config is for stuff that's constant on the field but this isn't. I'm not sure how much I like `FieldConfig` as it stands anyway.